### PR TITLE
Simplified MockSupportTest fixture in preparation for splitting

### PR DESCRIPTION
**Pull request by basvodde**
Now it'll be easier to split. Needs to be shrinked more still.

### CODE REVIEW

File: tests/CppUTestExt/MockSupportTest.cpp [hunk 1]
```diff
@@ -32,85 +32,67 @@
...
+  MockExpectedCallsListForTest expectations;
+  MockFailureReporterInstaller failureReporterInstaller;
+
+  void teardown()
+  {
+    mock().checkExpectations();
+    CHECK_NO_MOCK_FAILURE();
+  }
+};
+
```
Comment (arstrube) [inline tests/CppUTestExt/MockSupportTest.cpp]: Okay, that's less stuff. Do you propose duplicating this in every TEST_GROUP, or stuffing it into a fixture? Duplicate the fixutre in each file, or use a header?

(Same would wokr with TEST_BASE, of course :P)

Comment (basvodde) [inline tests/CppUTestExt/MockSupportTest.cpp]: You mean the 2 lines in teardown? Actually not all tests need these. Only the tests that check for MockFailure need the Reporter...

Comment (arstrube) [inline tests/CppUTestExt/MockSupportTest.cpp]: err, no. I mean everything. That includes the member variables. I mean, I am not dead against duplicating them. But we would need them once per test group (some maybe not), right?

Comment (basvodde) [inline tests/CppUTestExt/MockSupportTest.cpp]: Yes, you need them ones per testgroup. But not for all.

Comment (arstrube) [inline tests/CppUTestExt/MockSupportTest.cpp]: That wouldn't be too bad then. Also save us from creating a header.

**Comment by arstrube** [comment]
Looks good. If you want to merge it, I'll work from there.

**Comment by arstrube** [comment]
> Now it'll be easier to split. Needs to be shrinked more still.

Take it to a shrink, perhaps ~.~ ?

**Pull request merged**